src/onegov/ticket/upgrade.py

```diff
@@ -157,13 +154,16 @@ def add_closed_on_column_to_ticket(context: UpgradeContext) -> None:
         'tickets', Column('closed_on', UTCDateTime, nullable=True)
     )
 
-    for ticket in (context.session.query(Ticket).
-            filter(Ticket.state.in_(('closed', 'archived'))).
-            filter(Ticket.reaction_time.isnot(None)).
-            filter(Ticket.process_time.isnot(None))):
-
-        ticket.closed_on = (ticket.created +
-                            timedelta(seconds=ticket.reaction_time or 0) +
-                            timedelta(seconds=ticket.process_time or 0))
-
+    stmt = update(Ticket).where(
+        and_(
+            Ticket.state.in_(('closed', 'archived')),
+            Ticket.reaction_time.isnot(None),
+            Ticket.process_time.isnot(None)
+        )
+    ).values(
+        closed_on=Ticket.created +
+                  func.make_interval(secs=Ticket.reaction_time) +
+                  func.make_interval(secs=Ticket.process_time)
+    )
```

Comment:
This might actually only work in SQLAlchemy 2.0, I think the `select` API only worked for table objects in SQLAlchemy 1.3/1.4

So you would need to replace all instances of `Ticket` with `Ticket.__table__` and for the columns you would need to insert an additional `.c.` in between, so e.g. `Ticket.__table__.c.state`

Reply:
Locally it ran through without issues though, e.g. `onegov-core --select '/onegov_org/*' upgrade`